libs/langchain/langchain/vectorstores/neo4j_vector.py

```diff
@@ -0,0 +1,623 @@
+        username: str,
+        password: str,
+        url: str,
+        embedding_function: Embeddings,
+        database: str = "neo4j",
+        index_name: str = "vector",
+        node_label: str = "Chunk",
+        embedding_node_property: str = "embedding",
+        text_node_property: str = "text",
+        distance_strategy: DistanceStrategy = DEFAULT_DISTANCE_STRATEGY,
+        logger: Optional[logging.Logger] = None,
+        pre_delete_collection: bool = False,
+        relevance_score_fn: Optional[Callable[[float], float]] = None,
+    ) -> None:
+        try:
+            import neo4j
+        except ImportError:
+            raise ValueError(
+                "Could not import neo4j python package. "
+                "Please install it with `pip install neo4j`."
+            )
+
+        self._driver = neo4j.GraphDatabase.driver(url, auth=(username, password))
+        self._database = database
+        self.schema = ""
+        # Verify connection
+        try:
+            self._driver.verify_connectivity()
+        except neo4j.exceptions.ServiceUnavailable:
+            raise ValueError(
+                "Could not connect to Neo4j database. "
+                "Please ensure that the url is correct"
+            )
+        except neo4j.exceptions.AuthError:
+            raise ValueError(
+                "Could not connect to Neo4j database. "
+                "Please ensure that the username and password are correct"
+            )
+
+        # Verify if the version support vector index
+        self.verify_version()
+
+        self.embedding_function = embedding_function
+        self._distance_strategy = distance_strategy.value
+        self.index_name = index_name
+        self.node_label = node_label
+        self.embedding_node_property = embedding_node_property
+        self.text_node_property = text_node_property
+        self.logger = logger or logging.getLogger(__name__)
+        self.override_relevance_score_fn = relevance_score_fn
+
+        # Calculate embedding dimension
+        self.embedding_dimension = len(embedding_function.embed_query("foo"))
+
+        # Delete existing data if flagged
+        if pre_delete_collection:
+            self.query(f"MATCH (n:{self.node_label}) DETACH DELETE n")
+
+    def query(self, query: str, params: dict = {}) -> List[Dict[str, Any]]:
+        """
+        This method sends a Cypher query to the connected Neo4j database
+        and returns the results as a list of dictionaries.
+
+        Args:
+            query (str): The Cypher query to execute.
+            params (dict, optional): Dictionary of query parameters. Defaults to {}.
+
+        Returns:
+            List[Dict[str, Any]]: List of dictionaries containing the query results.
+
+        Raises:
+            ValueError: If the generated Cypher statement is not valid.
+        """
+        from neo4j.exceptions import CypherSyntaxError
+
+        with self._driver.session(database=self._database) as session:
```

Comment:
you can do this on the driver level with the current driver, no need for session anymore.
```python
records, summary, keys = driver.execute_query(
    "MATCH (p:Person {age: $age}) RETURN p.name AS name",
    age=42,
    database_="neo4j",
)
```